### ISSUE TITLE
SQLite like operator doesn't work with special symbols

### ISSUE THREAD

**Issue by dvladimirovich** [igor-tkachev/bltoolkit#399]
I have an app which works with SQLite as data store. There is an issue when I'm using context search. If I use simple phrases BLToolkit converts it in `LIKE '%<search string>%'` and everything works fine. But I want to use special characters as `_` to have an ability do search if I've forgotten what char must be here (e.g. when name is written in different languages). And when I print string like: `_ванов` in my search field BLToolkit generates next query:
[LastQuery.txt](https://github.com/igor-tkachev/bltoolkit/files/1040860/LastQuery.txt)
And it uses `ESCAPE~` for `_` symbol. As a result it can't find anything...
How could I resolve this problem?

**Comment by ili** [comment]
This by design, by default string.Contains() escapes special symbols, use Sql.Like() extension, it should not escape

**Comment by dvladimirovich** [comment]
There is a little problem. I'm using layered approach in my app, so UI layer knows nothing about others...
And I pass Expression tree predicate not simple sql

**Comment by dvladimirovich** [comment]
Is there some flag to turn off escaping? Or could you give me some advice on how to resolve this problem?

**Comment by ili** [comment]
@dvladimirovich heh... really have no other idea :(

**Comment by jogibear9988** [comment]
You could rewrite your Expression Tree befor handing over to linq2db. And replace every contains with Sql.like

**Comment by dvladimirovich** [comment]
Oh, I see...I suspected that I would have to do just that. And I did not want to do this, but I had to...

**Comment by jogibear9988** [comment]
i think with expressionvisitor it should not be a big task, see https://stackoverflow.com/questions/17728188/rewrite-expression-to-replace-list-contains-with-custom-method

**Comment by dvladimirovich** [comment]
Yeah, I've already had such experience in rewriting Expression trees using expression visitor=)
